### PR TITLE
[PRM-1005] Added model and enum for SDES Callback

### PR DESCRIPTION
**Pull request by ruthdas03**

### CODE REVIEW

File: app/models/SDESCallback.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                         checksum: Option[String],
+                         availableUntil: Option[LocalDateTime] = None,
+                         failureReason: Option[String],
+                         dateTime: Option[LocalDateTime] = None,
+                         properties: Option[Seq[Properties]]
+                       )
+case class Properties(name: String, value: String)
+
+object SDESCallback {
+  implicit val formatProperties: Reads[Properties] = Json.reads[Properties]
```
Comment (LewisMitchell) [inline app/models/SDESCallback.scala:38]: The reads for properties should be in its own companion object

Comment (LewisMitchell) [reply app/models/SDESCallback.scala:38]: i.e. 
```
object Properties {
  implicit val formatProperties: Reads[Properties] = Json.reads[Properties]
}
```

File: app/models/SDESFileNotificationEnum.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+package models
+
+import play.api.libs.json.{JsError, JsResult, JsSuccess, JsValue, Reads}
+
+object SDESFileNotificationEnum extends Enumeration {
+
+  val FileReady:SDESFileNotificationEnum.Value = Value
+  val FileReceived:SDESFileNotificationEnum.Value = Value
+  val FileProcessingFailure:SDESFileNotificationEnum.Value = Value
+  val FileProcessed:SDESFileNotificationEnum.Value = Value
```
Comment (LewisMitchell) [inline app/models/SDESFileNotificationEnum.scala:26]:
```suggestion
  val FileReady: SDESFileNotificationEnum.Value = Value
  val FileReceived: SDESFileNotificationEnum.Value = Value
  val FileProcessingFailure: SDESFileNotificationEnum.Value = Value
  val FileProcessed: SDESFileNotificationEnum.Value = Value
```

File: test/models/SDESCallbackSpec.scala [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    """
+      | {
+      |            "notification": "FileReady",
+      |            "filename": "axyz.doc",
+      |            "correlationID":"545d0831-d4ba-408d-b1f1-f4645efb32fd",
+      |            "dateTime": "2021-01-01T10:01:00.889Z"
+      |        }
+      |""".stripMargin)
+
+  val sdesCallBackModel:SDESCallback = SDESCallback(
```
Comment (LewisMitchell) [inline test/models/SDESCallbackSpec.scala:62]:
```suggestion
  val sdesCallBackModel: SDESCallback = SDESCallback(
```

File: test/models/SDESCallbackSpec.scala [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    checksum = Some("c6779ec2960296ed9a04f08d67f64422"),
+    correlationID = "545d0831-d4ba-408d-b1f1-f4645efb32fd",
+    availableUntil = Some(LocalDateTime.of(2021, 1, 6, 10, 1, 0).plus(889, ChronoUnit.MILLIS)),
+    failureReason = Some("Virus Detected"),
+    dateTime = Some(LocalDateTime.of(2021, 1, 1, 10, 1, 0).plus(889, ChronoUnit.MILLIS)),
+    properties = Some(Seq(Properties(name = "name1", value = "value1"),
+                          Properties(name = "name2", value = "value2")))
+  )
+
+  val sdesCallBackModelWithRequiredValues:SDESCallback = SDESCallback(
```
Comment (LewisMitchell) [inline test/models/SDESCallbackSpec.scala:75]:
```suggestion
  val sdesCallBackModelWithRequiredValues: SDESCallback = SDESCallback(
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks great. Just some minor things :)

File: app/models/SDESCallback.scala (hunk 1 above)
Comment (ruthdas03) [reply app/models/SDESCallback.scala:38]: > i.e.
> 
> ```
> object Properties {
>   implicit val formatProperties: Reads[Properties] = Json.reads[Properties]
> }
> ```

👍

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**